### PR TITLE
Using Parallel CSV Reader as a Default Option

### PR DESCRIPTION
**Pull request by pdet**
In this PR, we eliminate the `SET experimental_parallel_csv=true;` flag in order to use the Parallel CSV Reader. From now on, we will always use the Parallel CSV Reader, where possible.

The following situations will still be automatically executed with the Single Threaded CSV Reader. 
1. The delimiter, escape, or quote options are empty, null, or composed of multiple characters.
2. We have a mix of Newline Identifiers (e.g., the file has `\n` and `\r\n')
3. The `null_padding` option is set. (Obs: This PR also changed this option to false as default).
4. The CSV is being fed by a pipe.

I believe that 2. and 3. can not be implemented in our current Parallel CSV Reader design, with the guarantee that the result will all be correct. All the others should be possible and done incrementally.

For testing the Parallel CSV Default, I've created the `test/parallel_csv/test_parallel_csv.cpp` file.
It basically loops over all the CSV Files we have in our DuckDB repo and attempts to stress the breakpoints of the parallelism (i.e., buffer sizes and the number of threads executing on that buffer). For every file, it runs on different configurations of threads (From 1 to 8), and different buffer sizes (From the biggest line size to +5 of that, incremented by 1, on each run. Due to their time consumption, these tests are tagged as slow tests (hence only executed on Release CI builds).

Noticeable:
1. `parallel` option to `read_csv` and `read_csv_auto` that can be set to false to run the single-threaded scanner.
2. Difference between printing physical operators that run the parallel or the single-threaded scanner for optimizer tests.
3. The single-threaded scanner also captures the max line size of a file, that's only used for debugging.
4. The Parallel CSV Reader is not yet outputting the correct line number on a broken CSV File. This will come in a following PR (Hence the tests for these currently commented out).

### CODE REVIEW

File: src/function/table/read_csv.cpp [hunk 1]
```diff
@@ -23,21 +24,22 @@ unique_ptr<CSVFileHandle> ReadCSV::OpenCSV(const string &file_path, FileCompress
 	auto opener = FileSystem::GetFileOpener(context);
 	auto file_handle =
 	    fs.OpenFile(file_path.c_str(), FileFlags::FILE_FLAGS_READ, FileLockType::NO_LOCK, compression, opener);
+	if (fs.FileExists(file_path)) {
```
Comment (Mytherin) [inline src/function/table/read_csv.cpp:27]: Same here - this should probably check the `file_handle` properties

File: test/sql/copy/csv/csv_quoted_newline_incorrect.test [hunk 2]
```diff
@@ -7,13 +7,5 @@ require vector_size 512
 statement ok
 PRAGMA enable_verification
 
-query II
+statement maybe
```
Comment (Mytherin) [inline test/sql/copy/csv/csv_quoted_newline_incorrect.test:10]: Is this actually required to be `maybe` still?

File: test/sql/copy/csv/test_union_by_name.test [hunk 3]
```diff
@@ -63,8 +63,6 @@ BIGINT	BIGINT	BIGINT
 # Test timestamp type
 statement error
 SELECT * FROM read_csv_auto(['data/csv/union-by-name/ubn1.csv', 'data/csv/union-by-name/ubn2.csv', 'data/csv/union-by-name/ubn3.csv', 'data/csv/union-by-name/ubn4.csv'])
-----
-Could not convert string
```
Comment (Mytherin) [inline test/sql/copy/csv/test_union_by_name.test:67]: Does this error message need to be removed, or can it just be changed to something else?

File: test/sql/copy/csv/null_padding_big.test [hunk 4]
```diff
@@ -8,32 +8,30 @@ statement ok
...
 # one line that requires null padding
 query I
-SELECT COUNT(*) FROM read_csv_auto('data/csv/nullpadding_big_mixed.csv', null_padding=False, ignore_errors=True)
+SELECT COUNT(*) FROM read_csv_auto('data/csv/nullpadding_big_mixed.csv', null_padding=False, ignore_errors=True, parallel = false)
 ----
 2500
 
-query I
+statement error
 SELECT COUNT(*) FROM read_csv_auto('data/csv/nullpadding_big_mixed.csv', parallel=True)
```
Comment (Mytherin) [inline test/sql/copy/csv/null_padding_big.test:27]: Can we add an error message here, or is that random/non-deterministic currently?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Great work on this, super exciting! Some comments below:

**Comment by shreeve** [comment]
Quick question - For a pretty significant change like this one, what does a release schedule look like? Is there a release schedule URL or what's the best way to see when this might be released?

**Comment by pdet** [comment]
> Quick question - For a pretty significant change like this one, what does a release schedule look like? Is there a release schedule URL or what's the best way to see when this might be released?

Hi @shreeve, the next release is planned for ~18th of May, basically, all new features/bug-fixes merged into master before the 8th of May will be featured in the next release.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**